File: programs-basic-java/src/test/java/com/local/numbertypes/PalindromeNumberTest.java
package com.local.numbertypes;

import static com.local.numbertypes.NumberType.PALINDROME_NUMBER;

import java.util.List;

import org.junit.jupiter.api.Test;

class PalindromeNumberTest extends AbstractNumberTest {

	@Test
	void testPalindromeNumbers() {
		testNumbersTypeByFile(PALINDROME_NUMBER, "palindromenumbers.txt");
	}

	@Test
	void testPalindromeNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(PALINDROME_NUMBER, List.of(55, 66, 777), 3);
	}
}

File: programs-basic-kotlin/src/test/kotlin/com/local/numbertypes/PalindromeNumberTest.kt
package com.local.numbertypes
import org.junit.jupiter.api.Test
class PalindromeNumberTest : AbstractNumberTest() {
	@Test
	fun testPalindromeNumbers() {
		testNumbersTypeByFile(com.local.numbertypes.NumberType.PALINDROME_NUMBER, "palindromenumbers.txt")
	}

	@Test
	fun testPalindromeNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(com.local.numbertypes.NumberType.PALINDROME_NUMBER, java.util.List.of<Int>(55, 66, 777), 3)
	}
}